testutil/testmigration/fixtures_recoverable_account.go

```diff
@@ -0,0 +1,646 @@
+	return func(config *MorseFixturesConfig) {
+		config.ValidatorStakesConfigFn = cfg
+	}
+}
+
+// WithApplicationStakesFn sets the ApplicationStakesConfig for the fixtures.
+// It defines how staked and unstaked balances are determined for application accounts.
+func WithApplicationStakesFn(cfg ApplicationStakesConfigFn) MorseFixturesOption {
+	return func(config *MorseFixturesConfig) {
+		config.ApplicationStakesConfigFn = cfg
+	}
+}
+
+// NewMorseFixtures creates a new MorseMigrationFixtures instance with the provided options.
+// It initializes the necessary data structures and generates test fixtures according to
+// the configuration provided through the options.
+func NewMorseFixtures(opts ...MorseFixturesOption) (*MorseMigrationFixtures, error) {
+	morseFixtures := &MorseMigrationFixtures{
+		morseStateExport: &migrationtypes.MorseStateExport{
+			AppState: &migrationtypes.MorseTendermintAppState{
+				Application: &migrationtypes.MorseApplications{
+					Applications: make([]*migrationtypes.MorseApplication, 0),
+				},
+				Auth: &migrationtypes.MorseAuth{
+					Accounts: make([]*migrationtypes.MorseAuthAccount, 0),
+				},
+				Pos: &migrationtypes.MorsePos{
+					Validators: make([]*migrationtypes.MorseValidator, 0),
+				},
+			},
+		},
+		morseAccountState: &migrationtypes.MorseAccountState{
+			Accounts: make([]*migrationtypes.MorseClaimableAccount, 0),
+		},
+	}
+
+	// Apply all the provided functional options to configure the fixtures
+	morseFixtures.config = &MorseFixturesConfig{}
+	for _, opt := range opts {
+		opt(morseFixtures.config)
+	}
+
+	totalAccounts := morseFixtures.config.ValidAccountsConfig.NumAccounts +
+		morseFixtures.config.ValidAccountsConfig.NumApplications +
+		morseFixtures.config.ValidAccountsConfig.NumValidators +
+		morseFixtures.config.ValidAccountsConfig.NumModuleAccounts +
+		morseFixtures.config.InvalidAccountsConfig.NumAddressTooShort +
+		morseFixtures.config.InvalidAccountsConfig.NumAddressTooLong +
+		morseFixtures.config.InvalidAccountsConfig.NumNonHexAddress +
+		morseFixtures.config.OrphanedActorsConfig.NumApplications +
+		morseFixtures.config.OrphanedActorsConfig.NumValidators
+
+	morseFixtures.morseAccountState.Accounts = make([]*migrationtypes.MorseClaimableAccount, totalAccounts)
+
+	// Generate all the fixtures based on the configuration
+	if err := morseFixtures.generate(); err != nil {
+		return nil, err
+	}
+
+	return morseFixtures, nil
+}
+
+// GetConfig returns the configuration used for generating the fixtures.
+func (mf *MorseMigrationFixtures) GetConfig() *MorseFixturesConfig {
+	return mf.config
+}
+
+// NextIndex increments and returns the current index counter of the MorseMigrationFixtures.
+// This method is used to generate sequential indices for various entities created during the
+// fixture generation process, ensuring each entity has a unique index for deterministic generation.
+func (mf *MorseMigrationFixtures) NextIndex() uint64 {
```

Comment:
MUST this be exported?

And perhaps it should be renamed to `nextAllActorsIndex()`/`nextAllAccountsIndex()`